src/phpdoc/type_parser.go

```diff
@@ -129,20 +143,21 @@ const (
 )
 
 var prefixPrecedenceTab = [256]byte{
-	'?': 4,
-	'[': 4,
-	'!': 4,
+	'?': 5,
+	'[': 5,
+	'!': 5,
 }
 
 var infixPrecedenceTab = [256]byte{
 	':': 1,
 	'|': 2,
-	'&': 3,
-	'[': 4,
-	'<': 5,
-	'(': 5,
-	'{': 5,
-	'?': 4,
+	';': 3, // is ::
```

Comment:
Hmmm.. Clever. :joy: 
But maybe we need to reconsider this single-byte LUT.
Can you explain in parenthesis why `;` is `::`? It's not obvious until you scan across the parser code.